tiledb/sm/rest/rest_client.cc

```diff
@@ -674,6 +687,12 @@ Status RestClient::get_query_est_result_sizes(const URI& uri, Query* query) {
   return serialization::query_est_result_size_deserialize(
       query, serialization_type_, true, returned_data);
 }
+std::string RestClient::redirect_uri() {
+  std::unique_lock<std::mutex> rd_lck(redirect_mtx_);
+  std::unordered_map<std::string, std::string>::const_iterator got =
+      redirect_meta_.find(constants::redirection_header_key);
```

Comment:
This is keying on the redirection header key instead of the value. This should key on the entire array URI.